### PR TITLE
LowCost version

### PR DESCRIPTION
**Pull request by SimpleVlad**
### This pullrequest changes

So, this is a  low coast version openCV.js for specific tasks.
CC/ @dkurt

### CODE REVIEW

**Comment by SimpleVlad** [comment]
General configuration for OpenCV 4.1.1-dev =====================================
--   Version control:               4.1.1-157-gbe06763af
-- 
--   Platform:
--     Timestamp:                   2019-08-27T12:02:50Z
--     Host:                        Linux 5.0.0-25-generic x86_64
--     Target:                      Emscripten 1 x86
--     CMake:                       3.10.2
--     CMake generator:             Unix Makefiles
--     CMake build tool:            /usr/bin/make
--     Configuration:               Release
-- 
--   CPU/HW features:
--     Baseline:
-- 
--   C/C++:
--     Built as dynamic libs?:      NO
--     C++ Compiler:                /home/dizheninv/emsdk/fastcomp/emscripten/em++  (ver 6.0.1)
--     C++ flags (Release):         -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
--     C++ flags (Debug):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /home/dizheninv/emsdk/fastcomp/emscripten/emcc
--     C flags (Release):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
--     C flags (Debug):             -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):      -Wl,--gc-sections -O2 
--     Linker flags (Debug):        -Wl,--gc-sections  
--     ccache:                      YES
--     Precompiled headers:         NO
--     Extra dependencies:
--     3rdparty dependencies:       zlib libprotobuf
-- 
--   OpenCV modules:
--     To be built:                 calib3d core dnn features2d flann imgproc js objdetect photo video
--     Disabled:                    highgui imgcodecs java_bindings_generator ml python_bindings_generator stitching videoio world
--     Disabled by dependency:      -
--     Unavailable:                 gapi java python2 python3 ts
--     Applications:                -
--     Documentation:               js
--     Non-free algorithms:         NO
-- 
--   GUI: 
-- 
--   Media I/O: 
--     ZLib:                        build (ver 1.2.11)
--     HDR:                         YES
--     SUNRASTER:                   YES
--     PXM:                         YES
--     PFM:                         YES
-- 
--   Video I/O:
-- 
--   Parallel framework:            none
-- 
--   Other third-party libraries:
--     Custom HAL:                  NO
--     Protobuf:                    build (3.5.1)
-- 
--   Python (for build):            /usr/bin/python2.7
-- 
--   Install to:                    /home/dizheninv/opencv/build_js/install

**Comment by SimpleVlad** [comment]
General configuration for OpenCV 4.1.1-dev =====================================
   Version control:               4.1.1-158-gbefd36493
 
   Platform:
     Timestamp:                   2019-08-27T12:02:50Z
     Host:                        Linux 5.0.0-25-generic x86_64
     Target:                      Emscripten 1 x86
     CMake:                       3.10.2
     CMake generator:             Unix Makefiles
     CMake build tool:            /usr/bin/make
     Configuration:               Release
 
   CPU/HW features:
     Baseline:
 
   C/C++:
     Built as dynamic libs?:      NO
     C++ Compiler:                /home/dizheninv/emsdk/fastcomp/emscripten/em++  (ver 6.0.1)
     C++ flags (Release):         -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
     C++ flags (Debug):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
     C Compiler:                  /home/dizheninv/emsdk/fastcomp/emscripten/emcc
     C flags (Release):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
     C flags (Debug):             -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
     Linker flags (Release):      -Wl,--gc-sections -O2 
     Linker flags (Debug):        -Wl,--gc-sections  
     ccache:                      YES
     Precompiled headers:         NO
     Extra dependencies:
     3rdparty dependencies:       zlib libprotobuf
   OpenCV modules:
     To be built:                 core dnn flann imgproc js photo video
     Disabled:                    calib3d features2d highgui imgcodecs java_bindings_generator ml python_bindings_generator stitching videoio world
     Disabled by dependency:      objdetect
     Unavailable:                 gapi java python2 python3 ts
     Applications:                -
     Documentation:               js
     Non-free algorithms:         NO
 
   GUI: 
 
   Media I/O: 
     ZLib:                        build (ver 1.2.11)
     HDR:                         YES
     SUNRASTER:                   YES
     PXM:                         YES
     PFM:                         YES
-- 
   Video I/O:
 
   Parallel framework:            none
 
   Other third-party libraries:
--     Custom HAL:                  NO
--     Protobuf:                    build (3.5.1)
-- 
   Python (for build):            /usr/bin/python2.7
 
   Install to:                    /home/dizheninv/opencv/build_js/install
 -----------------------------------------------------------------

**Comment by SimpleVlad** [comment]
Not every modules are offline

**Comment by dkurt** [comment]
@SimpleVlad, please use ``` signs to format text. Like this:

```
    ```
    Other third-party libraries:
    Custom HAL: NO
    Protobuf: build (3.5.1)
    ```
```

For now it looks incomplete:

> Unavailable: gapi java python2 python3 ts
Applications: -
Documentation: js
Non-free algorithms: NO
> 
> GUI:
> 
> Media I/O:
ZLib: build (ver 1.2.11)
HDR: YES
SUNRASTER: YES
PXM: YES
PFM: YES

What about GUI? No info at all?

**Comment by SimpleVlad** [comment]
@dkurt, thanks. I haven't info about GUI. Maybe it's not working?

**Comment by dkurt** [comment]
@SimpleVlad, Got it. Current version is OK.

1) Please add current size to the latest message
2) Please try to disable protobuf
3) Check if list of modules is actual: `core dnn flann imgproc js photo video`

**Comment by SimpleVlad** [comment]
```
General configuration for OpenCV 4.1.1-dev =====================================
--   Version control:               4.1.1-159-ga6cf4d96a
-- 
--   Platform:
--     Timestamp:                   2019-08-29T13:28:20Z
--     Host:                        Linux 5.0.0-25-generic x86_64
--     Target:                      Emscripten 1 x86
--     CMake:                       3.10.2
--     CMake generator:             Unix Makefiles
--     CMake build tool:            /usr/bin/make
--     Configuration:               Release
-- 
--   CPU/HW features:
--     Baseline:
-- 
--   C/C++:
--     Built as dynamic libs?:      NO
--     C++ Compiler:                /home/dizheninv/emsdk/fastcomp/emscripten/em++  (ver 6.0.1)
--     C++ flags (Release):         -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
--     C++ flags (Debug):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /home/dizheninv/emsdk/fastcomp/emscripten/emcc
--     C flags (Release):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
--     C flags (Debug):             -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):      -Wl,--gc-sections -O2 
--     Linker flags (Debug):        -Wl,--gc-sections  
--     ccache:                      YES
--     Precompiled headers:         NO
--     Extra dependencies:
--     3rdparty dependencies:       zlib
-- 
--   OpenCV modules:
--     To be built:                 core flann imgproc js photo
--     Disabled:                    calib3d features2d highgui imgcodecs java_bindings_generator ml python_bindings_generator stitching video videoio world
--     Disabled by dependency:      objdetect
--     Unavailable:                 dnn gapi java python2 python3 ts
--     Applications:                -
--     Documentation:               js
--     Non-free algorithms:         NO
-- 
--   GUI: 
-- 
--   Media I/O: 
--     ZLib:                        build (ver 1.2.11)
--     HDR:                         NO
--     SUNRASTER:                   NO
--     PXM:                         NO
--     PFM:                         NO
-- 
--   Video I/O:
-- 
--   Parallel framework:            none
-- 
--   Other third-party libraries:
--     Custom HAL:                  NO
-- 
--   Python (for build):            /usr/bin/python2.7
```

File: platforms/js/build_js.py [hunk 1]
```diff
@@ -116,9 +116,9 @@ def get_cmake_cmd(self):
                "-DWITH_QUIRC=OFF",
                "-DBUILD_ZLIB=ON",
                "-DBUILD_opencv_apps=OFF",
-               "-DBUILD_opencv_calib3d=ON",
-               "-DBUILD_opencv_dnn=ON",
-               "-DBUILD_opencv_features2d=ON",
+               "-DBUILD_opencv_calib3d=OFF",
+               "-DBUILD_opencv_dnn=OFF",
+               "-DBUILD_opencv_features2d=OFF",
                "-DBUILD_opencv_flann=ON",  # No bindings provided. This module is used as a dependency for other modules.
```
Comment (dkurt) [inline platforms/js/build_js.py]: Please turn `flann` OFF as well

**Comment by SimpleVlad** [comment]
```
-- General configuration for OpenCV 4.1.1-dev =====================================
--   Version control:               4.1.1-160-g0f99efbc1
-- 
--   Platform:
--     Timestamp:                   2019-08-29T15:07:13Z
--     Host:                        Linux 5.0.0-25-generic x86_64
--     Target:                      Emscripten 1 x86
--     CMake:                       3.10.2
--     CMake generator:             Unix Makefiles
--     CMake build tool:            /usr/bin/make
--     Configuration:               Release
-- 
--   CPU/HW features:
--     Baseline:
-- 
--   C/C++:
--     Built as dynamic libs?:      NO
--     C++ Compiler:                /home/dizheninv/emsdk/fastcomp/emscripten/em++  (ver 6.0.1)
--     C++ flags (Release):         -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
--     C++ flags (Debug):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /home/dizheninv/emsdk/fastcomp/emscripten/emcc
--     C flags (Release):           -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG -O2  -DNDEBUG
--     C flags (Debug):             -s USE_PTHREADS=0   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):      -Wl,--gc-sections -O2 
--     Linker flags (Debug):        -Wl,--gc-sections  
--     ccache:                      YES
--     Precompiled headers:         NO
--     Extra dependencies:
--     3rdparty dependencies:       zlib
-- 
--   OpenCV modules:
--     To be built:                 core imgproc js photo
--     Disabled:                    calib3d features2d flann highgui imgcodecs java_bindings_generator ml python_bindings_generator stitching video videoio world
--     Disabled by dependency:      objdetect
--     Unavailable:                 dnn gapi java python2 python3 ts
--     Applications:                -
--     Documentation:               js
--     Non-free algorithms:         NO
-- 
--   GUI: 
-- 
--   Media I/O: 
--     ZLib:                        build (ver 1.2.11)
--     HDR:                         NO
--     SUNRASTER:                   NO
--     PXM:                         NO
--     PFM:                         NO
-- 
--   Video I/O:
-- 
--   Parallel framework:            none
-- 
--   Other third-party libraries:
--     Custom HAL:                  NO
-- 
--   Python (for build):            /usr/bin/python2.7
-- 
--   Install to:                    /home/dizheninv/opencv/build_js/install
```

**Pull request closed without merging**